app/routers/clash.py

```diff
@@ -0,0 +1,1177 @@
+        briefs.append(ClashProxyBriefResponse(
+            id=f"#{brief.id}",
+            name=brief.name,
+            tag=brief.tag,
+            server="",
+            builtin=brief.builtin
+        ))
+    
+    briefs.sort(key=lambda v: v.name)
+
+    return {"data": briefs}
+
+@router.get("/clash/proxy/tags", tags=['Clash'], response_model=ClashProxyTagsResponse)
+def get_clash_proxy_tags(db: Session = Depends(get_db),
+              admin: Admin = Depends(Admin.get_current)):
+    tags = []
+    cache: Dict[str, ClashProxyTagResponse] = {}
+
+    def add_tag(proxy, server):
+        tag = str(proxy.tag) or ""
+        tag_entry = cache.get(tag)
+        if tag and not tag_entry:
+            tag_entry = ClashProxyTagResponse(tag=tag, servers=[])
+            cache[tag] = tag_entry
+            tags.append(tag_entry)
+        tag_entry.servers.append(f"{proxy.name} -> {server}")
+
+    for proxy in crud.get_clash_proxies(db)[0]:
+        add_tag(proxy, proxy.server)
+
+    for proxy in crud.get_clash_proxy_groups(db)[0]:
+        if not proxy.builtin:
+            add_tag(proxy, "<Proxy Group>")
+    
+    tags.sort(key=lambda v: v.tag)
+
+    return {"data": tags}
+    
+@router.get("/clash/proxy/inbounds", tags=['Clash'], response_model=ClashProxyInboundsResponse)
+def get_clash_proxy_inbounds(db: Session = Depends(get_db),
+              admin: Admin = Depends(Admin.get_current)):
+    inbounds = []
+
+    for ib in xray.config.inbounds:
+        sni = ""
+        if ib["tls"] == "reality":
+            salt = secrets.token_hex(8)
+            sni = random.choice(ib["sni"]).replace('*', salt)
+
+        inbounds.append(ClashProxyInboundResponse(
+            name=ib["tag"],
+            server=SERVER_IP,
+            type=ib["protocol"],
+            security=ib["tls"] if ib["tls"] else "none",
+            network=ib["network"],
+            servername=sni,
+            port=ib["port"]
+        ))
+    
+    inbounds.sort(key=lambda v: v.name)
+
+    return {"data": inbounds}
+
+@router.get("/clash/settings", tags=['Clash'], response_model=ClashSettingsResponse)
+def get_clash_settings(
+             db: Session = Depends(get_db),
+             admin: Admin = Depends(Admin.get_current)):
+
+    data, count = crud.get_clash_settings(db=db)
+    
+    return {"data": data, "total": count}
+
+@router.put("/clash/setting/{name}", tags=['Clash'], response_model=ClashSettingResponse)
+def modify_clash_setting(name: str,
+                modified: ClashSettingCreate,
+                db: Session = Depends(get_db),
+                admin: Admin = Depends(Admin.get_current)):
+
+    dbsetting = crud.get_clash_setting(db=db, name=name)
+    if not dbsetting:
+        raise HTTPException404("Setting not found")
+
+    crud.update_clash_setting(db, dbsetting, modified)
+
+    logger.info(f"Setting \"{dbsetting.name}\" modified")
+
+    return dbsetting
+
+
+def generate_subscription(db: Session, dbuser: User, user_agent: str):
```

Comment:
we have a directory for subscription classes and generating them , why all of this cod this in routers ?
routers is just for api requests

Reply:
because it has a set of api to add, modify or delete proxy, proxy group, rule, ruleset or settings